File: BackEnd/BackEnd/Models/Movimientos.cs
namespace BackEnd.Models
{
  public class Movimientos
  {
    public int? Id { get; set; }
    public int? IdProducto { get; set; }
    public int? IdTipo { get; set; }
    public int? Cantidad { get; set; }
    public DateTime? Fecha { get; set; }
    public int? IdUsuario { get; set; }

    public TipoMovimiento? TipoMovimiento { get; set; }
  }
}

File: src/app/models/movimientos.ts
import { TipoMovimiento } from "./tipoMovimiento";

export class Movimientos{
    id?:number|null=null;
    idProducto?:number|null=null;
    idTipo?:number|null=null;
    cantidad?:number|null=null;
    fecha?:Date|null=null;
    idUsuario?:number|null=null;
    tipoMovimiento?:TipoMovimiento|null=null;
}